src/tests/multihost/alltests/test_proxy_rfc2307.py

```diff
@@ -0,0 +1,671 @@
+from sssd.testlib.common.expect import pexpect_ssh
+from constants import ds_suffix, ds_instance_name
+
+
+def execute_cmd(multihost, command):
+    """ Execute command on client """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.fixture(scope='class')
+def ldap_objects_sssd_client(multihost, request):
+    """
+        Configure sssd.conf
+        Create a dedicated user with
+        a DN starting e.g. with cn=...
+        Create a local user foo12
+    """
+    ldap_uri = 'ldap://%s' % multihost.master[0].sys_hostname
+    ds_rootdn = 'cn=Directory Manager'
+    ds_rootpw = 'Secret123'
+    ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+    user_info = {'objectClass': [b'top', b'organizationalUnit'],
+                 'ou': 'Users'.encode('utf-8')}
+    users_dn = 'ou=Users,dc=example,dc=test'
+    (_, _) = ldap_inst.add_entry(user_info, users_dn)
+    user_info = {'cn': 'User_CS1'.encode('utf-8'),
+                 'homeDirectory': '/home/User_CS1'.encode('utf-8'),
+                 'objectClass': [b'account',
+                                 b'posixAccount'],
+                 'uidNumber': '111111'.encode('utf-8'),
+                 'gidNumber': '111111'.encode('utf-8'),
+                 'loginShell': '/bin/bash'.encode('utf-8'),
+                 'userPassword': 'Secret123'.encode('utf-8'),
+                 'uid': ['User_CS1'.encode('utf-8'),
+                         'User_CS1_Alias'.encode('utf-8')]}
+    user_dn = f'uid=User_CS1,ou=Users,{ds_suffix}'
+    (_, _) = ldap_inst.add_entry(user_info, user_dn)
+    grop_info = {'cn': ['User_CS1_grp1'.encode('utf-8'),
+                        'User_CS1_grp1_Alias'.encode('utf-8')],
+                 'objectClass': b'posixGroup',
+                 'gidNumber': '111111'.encode('utf-8'),
+                 'memberUid': 'User_CS1'.encode('utf-8')}
+    group_dn = f'cn=User_CS1_grp1,ou=Groups,{ds_suffix}'
+    (_, _) = ldap_inst.add_entry(grop_info, group_dn)
+    execute_cmd(multihost, "> /etc/pam_ldap.conf")
+    execute_cmd(multihost, f"echo 'base {ds_suffix}' > "
+                           f"/etc/pam_ldap.conf")
+    execute_cmd(multihost, "echo 'pam_password md5'"
+                           " >> /etc/pam_ldap.conf")
+    execute_cmd(multihost, f"echo 'host {multihost.master[0].ip}' "
+                           f">> /etc/pam_ldap.conf")
+    execute_cmd(multihost, "echo 'tls_cacertfile "
+                           "/etc/openldap/certs/cacert.asc' >> "
+                           "/etc/pam_ldap.conf")
+    """
+        Interaction between nss-pam-ldapd and a feature of pam_usertype.
+        pam_usertype might send a lookup for a user called
+        'pam_usertype_non_existent:' and nss-pam-ldapd does not like the ':' in
+        the username.
+
+        To fix this we need to add following for RHEL9
+    """
+    version = "cat /etc/redhat-release | grep -o '[0-9.]\\+'"
+    if '9.' in execute_cmd(multihost, version).stdout_text:
```

Comment:
Hi,

would it be possible to add a check for Fedora as well and add the extended `validnames` if the test is running on Fedora as well?

bye,
Sumit

Reply:
Done

Reply:
Hi,

you've added `or 'Fedora'` but this won't work because the `grep` in the command will only return numbers and '.'. Why did you add the `grep` in the first place? I guess if you remove the `grep` it should work.

bye,
Sumit

Reply:
fixed now